### DISCUSSION TITLE
Request Enhancement: -extent

### DISCUSSION THREAD

**Discussion by fmw42** [ImageMagick/ImageMagick#4859] [Development]
I would like to be able to limit the -extent depending upon the image size using the ">" or "<" characters. 

For example:

```
-extent "256x256<"  would pad only if the image is smaller than 256x256
-extent "256x256>"  would crop only if the image is larger than 256x256

```

This would avoid having to write a ternary %[fx: ... ] expression. So it is a convenience enhancement.

**Comment by urban-warrior** [comment]
We'll support this enhancement in the Beta after tomorrow's release.

**Comment by fmw42** [comment]
Sorry, what I wrote above is not quite what I had meant. -extent already does what I wrote above.  

What I really meant was:

```
-extent "256x256<"  would (process) pad only if the image is smaller than 256x256, AND not crop if the image is larger (i.e. no-op).

-extent "256x256>"  would (process) crop only if the image is larger than 256x256, AND not extend if the image is smaller. (i.e. no-op).


```

**Reply by fmw42** [reply]
Basically, the use of < or > would allow -extent to limit the pad/crop to either some minimum or maximum output size.

```
256x256<  would indicate the minimum size (pad to that size if image is smaller))
256x256>  would indicate the maximum size (crop to that size if image is larger)
```

**Comment by fmw42** [comment]
The latest beta is working correctly, now, thanks.
